components/rustlib/rust-toolchain.toml

```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "esp"
```

Comment:
This should be reverted. Our esp toolchain supports both Xtensa & RISCV chips but upstream nightly only supports RISCV

Reply:
I wouldn't touch this place, but I can't build it with "esp". `error: invalid toolchain name: 'esp'`

```
$ idf.py build
Executing action: all (aliases: build)
Running cmake in directory /user/p/rust/rust-esp32-example/build
Executing "cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DPYTHON=/home/mak/.espressif/python_env/idf5.1_py3.10_env/bin/python -DESP_PLATFORM=1 -DIDF_TARGET=esp32c3 -DCCACHE_ENABLE=0 /user/p/rust/rust-esp32-example"...
-- Found Git: /usr/bin/git (found version "2.35.3") 
-- The C compiler identification is GNU 11.2.0
-- The CXX compiler identification is GNU 11.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/mak/.espressif/tools/riscv32-esp-elf/esp-2022r1-11.2.0/riscv32-esp-elf/bin/riscv32-esp-elf-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/mak/.espressif/tools/riscv32-esp-elf/esp-2022r1-11.2.0/riscv32-esp-elf/bin/riscv32-esp-elf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/mak/.espressif/tools/riscv32-esp-elf/esp-2022r1-11.2.0/riscv32-esp-elf/bin/riscv32-esp-elf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32c3
-- Project sdkconfig file /user/p/rust/rust-esp32-example/sdkconfig
Loading defaults file /user/p/rust/rust-esp32-example/sdkconfig.defaults...
/user/p/rust/rust-esp32-example/sdkconfig:734 CONFIG_ESP_TASK_WDT was replaced with CONFIG_ESP_TASK_WDT_INIT
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of time_t
-- Check size of time_t - done
-- Found Python3: /home/mak/.espressif/python_env/idf5.1_py3.10_env/bin/python (found version "3.10.5") found components: Interpreter 
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS - Success
-- App "esp32-hello-rust" version: de8ed28-dirty
-- Adding linker script /user/p/rust/rust-esp32-example/build/esp-idf/esp_system/ld/memory.ld
-- Adding linker script /user/p/esp-idf/components/esp_system/ld/esp32c3/sections.ld.in
-- Adding linker script /user/p/esp-idf/components/esp_rom/esp32c3/ld/esp32c3.rom.ld
-- Adding linker script /user/p/esp-idf/components/esp_rom/esp32c3/ld/esp32c3.rom.api.ld
-- Adding linker script /user/p/esp-idf/components/esp_rom/esp32c3/ld/esp32c3.rom.libgcc.ld
-- Adding linker script /user/p/esp-idf/components/esp_rom/esp32c3/ld/esp32c3.rom.newlib.ld
-- Adding linker script /user/p/esp-idf/components/esp_rom/esp32c3/ld/esp32c3.rom.version.ld
-- Adding linker script /user/p/esp-idf/components/esp_rom/esp32c3/ld/esp32c3.rom.eco3.ld
-- Adding linker script /user/p/esp-idf/components/soc/esp32c3/ld/esp32c3.peripherals.ld
-- Components: app_trace app_update bootloader bootloader_support bt clib cmock console cxx driver efuse esp-tls esp_adc esp_app_format esp_common esp_eth esp_event esp_gdbstub esp_hid esp_http_client esp_http_server esp_https_ota esp_https_server esp_hw_support esp_lcd esp_local_ctrl esp_netif esp_partition esp_phy esp_pm esp_psram esp_ringbuf esp_rom esp_system esp_timer esp_wifi espcoredump esptool_py fatfs freertos hal heap http_parser idf_test ieee802154 json log lwip main mbedtls mqtt newlib nvs_flash openthread partition_table protobuf-c protocomm pthread riscv rustlib sdmmc soc spi_flash spiffs tcp_transport ulp unity usb vfs wear_levelling wifi_provisioning wpa_supplicant
-- Component paths: /user/p/esp-idf/components/app_trace /user/p/esp-idf/components/app_update /user/p/esp-idf/components/bootloader /user/p/esp-idf/components/bootloader_support /user/p/esp-idf/components/bt /user/p/rust/rust-esp32-example/components/clib /user/p/esp-idf/components/cmock /user/p/esp-idf/components/console /user/p/esp-idf/components/cxx /user/p/esp-idf/components/driver /user/p/esp-idf/components/efuse /user/p/esp-idf/components/esp-tls /user/p/esp-idf/components/esp_adc /user/p/esp-idf/components/esp_app_format /user/p/esp-idf/components/esp_common /user/p/esp-idf/components/esp_eth /user/p/esp-idf/components/esp_event /user/p/esp-idf/components/esp_gdbstub /user/p/esp-idf/components/esp_hid /user/p/esp-idf/components/esp_http_client /user/p/esp-idf/components/esp_http_server /user/p/esp-idf/components/esp_https_ota /user/p/esp-idf/components/esp_https_server /user/p/esp-idf/components/esp_hw_support /user/p/esp-idf/components/esp_lcd /user/p/esp-idf/components/esp_local_ctrl /user/p/esp-idf/components/esp_netif /user/p/esp-idf/components/esp_partition /user/p/esp-idf/components/esp_phy /user/p/esp-idf/components/esp_pm /user/p/esp-idf/components/esp_psram /user/p/esp-idf/components/esp_ringbuf /user/p/esp-idf/components/esp_rom /user/p/esp-idf/components/esp_system /user/p/esp-idf/components/esp_timer /user/p/esp-idf/components/esp_wifi /user/p/esp-idf/components/espcoredump /user/p/esp-idf/components/esptool_py /user/p/esp-idf/components/fatfs /user/p/esp-idf/components/freertos /user/p/esp-idf/components/hal /user/p/esp-idf/components/heap /user/p/esp-idf/components/http_parser /user/p/esp-idf/components/idf_test /user/p/esp-idf/components/ieee802154 /user/p/esp-idf/components/json /user/p/esp-idf/components/log /user/p/esp-idf/components/lwip /user/p/rust/rust-esp32-example/main /user/p/esp-idf/components/mbedtls /user/p/esp-idf/components/mqtt /user/p/esp-idf/components/newlib /user/p/esp-idf/components/nvs_flash /user/p/esp-idf/components/openthread /user/p/esp-idf/components/partition_table /user/p/esp-idf/components/protobuf-c /user/p/esp-idf/components/protocomm /user/p/esp-idf/components/pthread /user/p/esp-idf/components/riscv /user/p/rust/rust-esp32-example/components/rustlib /user/p/esp-idf/components/sdmmc /user/p/esp-idf/components/soc /user/p/esp-idf/components/spi_flash /user/p/esp-idf/components/spiffs /user/p/esp-idf/components/tcp_transport /user/p/esp-idf/components/ulp /user/p/esp-idf/components/unity /user/p/esp-idf/components/usb /user/p/esp-idf/components/vfs /user/p/esp-idf/components/wear_levelling /user/p/esp-idf/components/wifi_provisioning /user/p/esp-idf/components/wpa_supplicant
-- Configuring done
-- Generating done
-- Build files have been written to: /user/p/rust/rust-esp32-example/build
Running ninja in directory /user/p/rust/rust-esp32-example/build
Executing "ninja all"...
[19/826] Generating ../../partition_table/partition-table.binPartition table binary generated. Contents:
*******************************************************************************
# ESP-IDF Partition Table
# Name, Type, SubType, Offset, Size, Flags
nvs,data,nvs,0x9000,24K,
phy_init,data,phy,0xf000,4K,
factory,app,factory,0x10000,1M,
*******************************************************************************
[573/826] Performing configure step for 'rustlib_project'FAILED: esp-idf/rustlib/stamp/rustlib_project-configure /user/p/rust/rust-esp32-example/build/esp-idf/rustlib/stamp/rustlib_project-configure 
cd /user/p/rust/rust-esp32-example/components/rustlib && /usr/bin/cmake -E env CARGO_BUILD_TARGET=riscv32imc-esp-espidf CARGO_BUILD_TARGET_DIR=/user/p/rust/rust-esp32-example/build/esp-idf/rustlib/target cargo clean && /usr/bin/cmake -E touch /user/p/rust/rust-esp32-example/build/esp-idf/rustlib/stamp/rustlib_project-configure
error: invalid toolchain name: 'esp'
[583/826] Performing configure step for 'bootloader'-- Found Git: /usr/bin/git (found version "2.35.3") .c.obj
-- The C compiler identification is GNU 11.2.0
-- The CXX compiler identification is GNU 11.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/mak/.espressif/tools/riscv32-esp-elf/esp-2022r1-11.2.0/riscv32-esp-elf/bin/riscv32-esp-elf-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/mak/.espressif/tools/riscv32-esp-elf/esp-2022r1-11.2.0/riscv32-esp-elf/bin/riscv32-esp-elf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/mak/.espressif/tools/riscv32-esp-elf/esp-2022r1-11.2.0/riscv32-esp-elf/bin/riscv32-esp-elf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32c3
-- Project sdkconfig file /user/p/rust/rust-esp32-example/sdkconfig
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of time_t
-- Check size of time_t - done
-- Adding linker script /user/p/esp-idf/components/soc/esp32c3/ld/esp32c3.peripherals.ld
-- App "bootloader" version: v5.1-dev-1793-gb5f69e5fb7
-- Adding linker script /user/p/esp-idf/components/esp_rom/esp32c3/ld/esp32c3.rom.ld
-- Adding linker script /user/p/esp-idf/components/esp_rom/esp32c3/ld/esp32c3.rom.api.ld
-- Adding linker script /user/p/esp-idf/components/esp_rom/esp32c3/ld/esp32c3.rom.libgcc.ld
-- Adding linker script /user/p/esp-idf/components/esp_rom/esp32c3/ld/esp32c3.rom.newlib.ld
-- Adding linker script /user/p/esp-idf/components/bootloader/subproject/main/ld/esp32c3/bootloader.ld
-- Adding linker script /user/p/esp-idf/components/bootloader/subproject/main/ld/esp32c3/bootloader.rom.ld
-- Components: bootloader bootloader_support efuse esp_app_format esp_common esp_hw_support esp_rom esp_system esptool_py freertos hal log main micro-ecc newlib partition_table riscv soc spi_flash
-- Component paths: /user/p/esp-idf/components/bootloader /user/p/esp-idf/components/bootloader_support /user/p/esp-idf/components/efuse /user/p/esp-idf/components/esp_app_format /user/p/esp-idf/components/esp_common /user/p/esp-idf/components/esp_hw_support /user/p/esp-idf/components/esp_rom /user/p/esp-idf/components/esp_system /user/p/esp-idf/components/esptool_py /user/p/esp-idf/components/freertos /user/p/esp-idf/components/hal /user/p/esp-idf/components/log /user/p/esp-idf/components/bootloader/subproject/main /user/p/esp-idf/components/bootloader/subproject/components/micro-ecc /user/p/esp-idf/components/newlib /user/p/esp-idf/components/partition_table /user/p/esp-idf/components/riscv /user/p/esp-idf/components/soc /user/p/esp-idf/components/spi_flash
-- Configuring done
-- Generating done
-- Build files have been written to: /user/p/rust/rust-esp32-example/build/bootloader
ninja: build stopped: subcommand failed.
ninja failed with exit code 1, output of the command is in the /user/p/rust/rust-esp32-example/build/log/idf_py_stderr_output_424918 and /user/p/rust/rust-esp32-example/build/log/idf_py_stdout_output_424918

```

Reply:
Did you install our toolchain from https://github.com/esp-rs/rust-build? Do you also have `rustup` installed on your system?

Reply:
>  have rustup - yes

Reply:
`rustup` had beed installed. Regarding https://github.com/esp-rs/rust-build - I now realize that this is most likely the reason...  I took through _ESP32-C3 Setup_ branch from readme, `rustup target add riscv32i-unknown-none-elf`, and didn't install esp-rs/rust-build toolchain... I'll check if it helps, thanks!

Reply:
Fixed. JFYI my way was literally following directions from https://github.com/esp-rs/esp-idf-template . `cargo install espup && espup install` etc...